### DISCUSSION TITLE
How to disable base-64 encoding of request body and response body when using the runner

### DISCUSSION THREAD

**Discussion by mardukbp** [schemathesis/schemathesis#2141] [Q&A] [answered]
I'm executing the runner with

```python
runner = schemathesis.runner.from_schema(
    schema
)
```

I would like to see the request body and response body when a test fails, but it is base64-encoded.

Alternatively, how can I generate a cassette without using the CLI?

Thank you in advance.

**Comment by Stranger6667** [comment, marked as answer]
I added some helper methods for payload decoding in #2163 (`deserialize_body`)

Hm, there is no direct integration between `CassetteWriter` and the runner :(
It is mostly a matter of refactoring so they can be used together

**Comment by mardukbp** [comment]
Awesome! Thanks a lot for adding those methods.
